Frontend/src/lib/dates.js

```diff
@@ -1,23 +1,33 @@
-import moment from 'moment'
+// start/end dates may have different time-of-days
+export const getDatesBetweenInclusive = (startDate, endDate, timeZone) => {
+  // avoid infinite loop on invalid params
+  if (startDate > endDate) return []
 
-export const getDatesStartingOn = (startDate, numberOfDays, options) => {
-  const { offset } = options || { offset: 0 }
-  const range = []
-  for (let i = offset; i < numberOfDays + offset; i++) {
-    range.push(moment(startDate).add(i, 'days').toDate())
+  const dates = []
+  let nextDate = new Date(startDate)
+  while (!areOnSameDate(nextDate, new Date(endDate), timeZone)) {
+    dates.push(nextDate)
+    nextDate = new Date(nextDate)
+    nextDate.setDate(nextDate.getDate() + 1)
   }
-  return range
+  dates.push(nextDate)
+  return dates
+}
+
+export const areOnSameDate = (date1, date2, timeZone) => {
```

Comment:
Why not use moment's isSame function?

Reply:
I thought we didn't want to use `moment`. I forget what alternative Gregor recommended, but I can look into that if you want.

Reply:
https://moment.github.io/luxon/#/ luxon is what he mentioned

Reply:
I will migrate all our moment things to luxon tomorrow or friday, so if you want you can already use it

Reply:
don't forget that you have to use `--build ` in the docker compose up statement when adding a new library

Reply:
It should be
```
return date1.setZone(timeZone).hasSame(date2.setZone(timeZone), 'day')
```
as long as the parameters are first converted to luxon `DateTime`s.
[`hasSame`](https://moment.github.io/luxon/api-docs/index.html#datetimehassame), [`setZone`](https://moment.github.io/luxon/api-docs/index.html#datetimesetzone)

Might wait until you're done the migration, since I'm a bit busy now anyway. Or you could merge this now and handle this change as part of the migration. (This `areOnSameDate` comparison of dates isn't new to this PR, it was already in the codebase before, just not as a separate function.)

Reply:
Done, should be good to go.